src/Components/Components/src/NavigationManager.cs

```diff
@@ -203,7 +203,15 @@ public virtual void Refresh(bool forceReload = false)
 
     private void NotFoundCore()
     {
-        _notFound?.Invoke(this, new NotFoundEventArgs());
+        if (_notFound == null)
+        {
+            // global router doesn't exist, no events were registered
+            NavigateTo($"{BaseUri}not-found");
+        }
+        else
+        {
+            _notFound.Invoke(this, new NotFoundEventArgs());
```

Comment:
Didn't realize this, but since `NotFoundEventArgs` doesn't have any arguments, we can reuse the same instance privately (like Array.Empty()`) and save allocations